test/server/drain_manager_impl_test.cc

```diff
@@ -151,6 +155,42 @@ TEST_P(DrainManagerImplTest, DrainDeadlineProbability) {
   }
 }
 
+TEST_P(DrainManagerImplTest, OnDrainCallbacks) {
+  const bool drain_gradually = GetParam();
+  ON_CALL(server_.options_, drainStrategy())
+      .WillByDefault(Return(drain_gradually ? Server::DrainStrategy::Gradual
+                                            : Server::DrainStrategy::Immediate));
+  ON_CALL(server_.options_, drainTime()).WillByDefault(Return(std::chrono::seconds(4)));
+
+  DrainManagerImpl drain_manager(server_, envoy::config::listener::v3::Listener::DEFAULT);
+
+  // Register callbacks (store in array to keep in scope for test)
+  std::array<testing::MockFunction<void(std::chrono::milliseconds)>, 20> cbs;
+  std::array<Common::CallbackHandlePtr, 20> cb_handles;
+  for (auto i = 0; i < 20; i++) {
+    auto& cb = cbs[i];
+    if (drain_gradually) {
+      auto step = 1000 / 20;
```

Comment:
Would be good to cover the following test cases:
- number of callbacks does not evenly divide 1000msec, example: 30 callbacks
- number of callbacks is larger than 1000, so the step returned by this computation is 0.

Reply:
- [ ] Add additional tests